### PR TITLE
Add integration tests for the multithreaded server and change the makefile to have that integration test.

### PR DESCRIPTION
**Pull request by adiraju13**
@RyanLPeterman @zylava please check out the changes I made for the new integration tests.

### CODE REVIEW

File: test/integration_multithread_test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	TCP_IP = '127.0.0.1'
+	TCP_PORT = 2020
+	BUFFER_SIZE = 1024
+	MESSAGE = "GET /echo HTTP/1.1"
+
+	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+	s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+	s.connect(('localhost', TCP_PORT))
+	s.send(MESSAGE)
+	time.sleep(5)
```
Comment (ryanlpeterman) [inline test/integration_multithread_test.py]: why not just use blocking handler? sleeps are a little less robust in the event that code you expect to run instantly ends up taking longer than the sleep due to some blocking. very unlikely i do admit, but i believe people usually stay away from sleeps for testing

File: test/integration_multithread_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	s.close()
+
+	if (data != "HTTP/1.1 200 OK\r\nContent-Length: 18\r\nContent-Type: text/plain\r\n\r\nGET /echo HTTP/1.1\r\n\r\n"):
+	    	print "Failure response: ", repr(data)
+                error_code[0] = 1
+		return
+
+	error_code[0] = 0
+
+def client_thread(args):
```
Comment (ryanlpeterman) [inline test/integration_multithread_test.py]: Aside from the sleep and message portion, these functions are identical. What do you think about deleting one of the two and just passing in the message to each. The main thing we need to check for in the response is 200OK which could be shared among the two requests. If anything you can have the function return the data and check it in main so you dont have to copy paste the function code.

File: test/integration_multithread_test.py (hunk 1 above)
Comment (adiraju13) [reply test/integration_multithread_test.py]: How would I use blocking handler?

Comment (adiraju13) [reply test/integration_multithread_test.py]: jk I got it

**Comment by ryanlpeterman** [comment]
Looks good to me, feel free to merge

**Comment by ryanlpeterman** [comment]
Hi. if you're grading this, we were told by Shaghayegh, our TA, to include the conversations we had in messenger to count for credit about the code since we were getting lower grades on the code review comments:
selection_076

**Pull request merged**